components/script/dom/mutationrecord.rs

```diff
@@ -0,0 +1,135 @@
+use core::ptr;
+use dom_struct::dom_struct;
+use dom::bindings::codegen::Bindings::MutationRecordBinding::MutationRecordBinding::MutationRecordMethods;
+use dom::bindings::js::{JS, Root};
+use dom::bindings::str::DOMString;
+use dom::bindings::reflector::Reflector;
+use dom::node::Node;
+use dom::nodelist::NodeList;
+use dom::window::Window;
+use std::default::Default;
+
+#[dom_struct]
+pub struct MutationRecord {
+  reflector_: Reflector,
+
+//  readonly attribute DOMString record_type;
+  record_type: DOMString,
+
+//  [SameObject]
+//  readonly attribute Node target;
+  target: Root<Node>,
+
+//  [SameObject]
+//  readonly attribute NodeList addedNodes;
+  addedNodes: Root<NodeList>,
+
+//  [SameObject]
+//  readonly attribute NodeList removedNodes;
+  removedNodes: Root<NodeList>,
+
+//  readonly attribute Node? previousSibling;
+  previousSibling: Root<Node>,
+
+//  readonly attribute Node? nextSibling;
+  nextSibling: Root<Node>,
+
+//  readonly attribute DOMString? attributeName;
+  attributeName: DOMString,
+
+//  readonly attribute DOMString? attributeNamespace;
+  attributeNamespace: DOMString,
+
+//  readonly attribute DOMString? oldValue;
+  oldValue: DOMString,
```

Comment:
Given my request to remove the methods we don't implement yet, let's remove all of these members except for `record_type` and `target`.